Add Enterprise LDAP group links

zoidbergwill opened the pull request:
Closes https://github.com/xanzy/go-gitlab/issues/605

svanharmelen commented on groups.go:
```diff
@@ -48,6 +48,15 @@ type Group struct {
+	LDAPCN               string             `json:"ldap_cn"`
+	LDAPAccess           int64              `json:"ldap_access"`
+	LDAPGroupLinks       []LDAPGroupLink    `json:"ldap_group_links"`
+}
```
Can you point me to something that shows there fields are part of the response when a LDAP group is linked? If there isn't something about this in the docs, maybe the raw JSON API response? Thanks!

zoidbergwill replied:
Here's an example for an example listing of the links

```
[
  {
    "cn": "gitlab_group_example_30",
    "group_access": 30,
    "provider": "example_ldap_provider"
  },
  {
    "cn": "gitlab_group_example_40",
    "group_access": 40,
    "provider": "example_ldap_provider"
  }
]
```

benoitmenard-richemont commented:
Hi,

When do you think this MR will be merged ?

Thanks a lot,
Benoit

svanharmelen commented:
Well it looks like @zoidbergwill has abandoned the PR... I can maybe have a look at tweaking the PR myself, but sure when I'll have some time for that.

It will help if you can answer the last question about some docs or a raw API response that shows that the fields added in this PR are actually there...

benoitmenard-richemont commented:
There is no docs about this on Gitlab.com, but we use Gitlab EE so I can get the JSON response if you want.

I will start a new PR if you want.

svanharmelen commented:
We can use this PR (I can update it), so that's fine... Having a JSON response would  be great 👍

zoidbergwill commented:
I can update it today. Didn't see this in my email. It's been a long while since I opened it.

svanharmelen commented:
Cool, that would be very nice @zoidbergwill. Thanks!

zoidbergwill commented:
@benoitmenard-richemont I'm no longer on the team that manages GitLab at my company, so would you be able to test whether creating an LDAP Group Link returns the single link? vs a list? and what response deleting a link returns? 

Other than that I'm confident with my changes, going with assuming that they return the same things as other API endpoints of the GitLab API.

I guess I could check the GitLab codebase.

Then we can add tests for adding and deleting, and then I'm happy with addressing all the feedback.

jasoncarlsen commented:
@zoidbergwill I'm an admin of my company's GitLab instance so I can help answer the API response questions.  When deleting an LDAP group link there is no content in the response (204).  The response when creating an LDAP group link is:

```
{
  'cn': 'group_that_was_added',
  'group_access': 40,
  'provider': 'provider_for_group_that_was_added'
}
```

zoidbergwill commented on groups.go:
```diff
@@ -347,3 +354,86 @@ func (s *GroupsService) ListSubgroups(gid interface{}, opt *ListSubgroupsOptions
+		u = fmt.Sprintf("%s/%s", u, opt.Provider)
+	}
+	u = fmt.Sprintf("%s/%s", u, opt.CN)
+	req, err := s.client.NewRequest("POST", u, opt, options)
```
```suggestion
	req, err := s.client.NewRequest("DELETE", u, opt, options)
```
